packages/desktop-client/src/components/manager/subscribe/common.js

```diff
@@ -46,6 +45,26 @@ export function useBootstrapped() {
         } else {
           ensure('/bootstrap');
         }
+      };
+
+      let url = await send('get-server-url');
+      if (url == null) {
+        // A server hasn't been specified yet
+        try {
+          let response = await fetch('/account/needs-bootstrap');
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏we already had the API call here before (`send("subscribe-needs-bootstrap")`). We can keep the same logic there and make just a single API call. Just need to re-arrange a few lines..

IMO this should work:

```javascript
let { error, bootstrapped } = await send('subscribe-needs-bootstrap');
if (error) {
  let url = await send('get-server-url');
  if (url == null) {
    // A server hasn't been specified yet
    history.push('/config-server');
  } else {
    history.push('/error', { error });
  }
} else if (bootstrapped) {
  ensure('/login');
} else {
  ensure('/bootstrap');
}
```

Reply:
Just got the chance to look at this. I don’t think this code will work. By default, the server URL is not configured, so `subscribe-needs-bootstrap` will return `{ bootstrapped: true }`. We specifically want to call the `/account/needs-bootstrap` route on the _current domain_ rather than the one for any configured server. (Your proposed code also doesn’t have any logic to change the current server)

Reply:
Good point. Would passing the `serverUrl` as a argument of `send('subscribe-needs-bootstrap')` solve the problem?

Reply:
Let me know what you think!